### PR TITLE
Extract pushToRemote in scaffolder-backend

### PR DESCRIPTION
**Pull request by mfrinnstrom**
## Hey, I just made a Pull Request!

I intended to start working on #3138 but decided to do this before to make it a bit easier to follow the test code and not have the functionality duplicated across the different publishers.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/stages/publish/github.test.ts [hunk 1]
```diff
@@ -72,8 +56,13 @@ describe('GitHub Publisher', () => {
             clone_url: 'mockclone',
           },
         } as OctokitResponse<ReposCreateInOrgResponseData>);
+        mockGithubClient.users.getByUsername.mockResolvedValue({
+          data: {
+            type: 'Organization',
+          },
+        } as OctokitResponse<UsersGetByUsernameResponseData>);
```
Comment (mfrinnstrom) [inline plugins/scaffolder-backend/src/scaffolder/stages/publish/github.test.ts:63]: I had to add this to get the test to pass. Not sure how it passed before..

**Review by Rugvip — APPROVED** [review]
Awesome refactor, thanks! 😁

**Comment by mfrinnstrom** [comment]
@Rugvip Thanks. How do I get this merged though? 😄

**Comment by Rugvip** [comment]
@mfrinnstrom Just giving other people some time to chime in :grin:

**Pull request merged**